DependencyInjection/CmfBlogExtension.php

```diff
@@ -35,53 +35,57 @@ public function load(array $configs, ContainerBuilder $container)
         $config = $this->processConfiguration($configuration, $configs);
 
         $loader = new XmlFileLoader($container, new FileLocator(__DIR__.'/../Resources/config'));
-        $loader->load('controllers.xml');
+        $loader->load('services.xml');
 
-        if ($config['use_sonata_admin']) {
+        if (isset($config['persistence']['phpcr'])) {
+            $this->loadPhpcrPersistence($config, $loader, $container);
+        }
+
+        if (isset($config['sonata_admin']) && $config['sonata_admin']['enabled']) {
             $this->loadSonataAdmin($config, $loader, $container);
         }
-        if ($config['integrate_menu']['enabled']) {
+
+        if (isset($config['integrate_menu']) && $config['integrate_menu']['enabled']) {
             $this->loadMenuIntegration($config, $loader, $container);
```

Comment:
Should probably pass `$config['integrate_menu']` here instead of the whole configuration array.